infrastructure/general/setup-secure-wss.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Setup Secure WebSocket
+description: Instructions on setting up a secure socket for remote connections.
+---
+
+<!-- TODO: link relevant guides (bootnode, rpc) -->
+
+## Secure a WebSocket Port
+
+A non-secure websocket port can be converted to a secure WSS port by placing it behind an SSL-enabled proxy. This can be used to secure a bootnode or secure a RPC server. The SSL-enabled apache2/nginx/other proxy server redirects requests to the internal websocket and converts it to a secure (WSS) connection. For this, you will need an SSL certificate for which you can use a service like LetsEncrypt or self-signing.
+
+### Obtaining an SSL Certificate
+
+One easy way to get a free SSL certificate can be achieved by following the LetsEncrypt instructions
+([nginx](https://certbot.eff.org/instructions?ws=nginx&os=ubuntufocal){target=_blank}/[apache](https://certbot.eff.org/instructions?ws=apache&os=ubuntufocal){target=_blank}).
+This will auto-generate an SSL certificate and include it in your configuration.
+
+Alternatively, you can generate a self-signed certificate and rely on the raw IP address of your
+node when connecting to it. This isn't preferable since you will have to whitelist the certificate
+to access it from a browser.
+
+```bash
+sudo openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/selfsigned.key -out /etc/ssl/certs/selfsigned.crt
+sudo openssl dhparam -out /etc/ssl/certs/dhparam.pem 2048
+```
+
+## Installing a Proxy Server
+
+There are a lot of different implementations of a websocket proxy, some of the more widely used are
+[nginx](https://www.nginx.com/){target=_blank} and [apache2](https://httpd.apache.org/){target=_blank}, for which configuration
+examples provided below.
+
+### NGINX
+
+```bash
+apt install nginx
+```
```

Comment:
Probably this is super nit-picking (sorry for that!), but I think it would be helpful to add a small intro like: "To install NGINX on your machine, follow these steps:" and also enumerate those steps:

```suggestion
1. Install the Nginx web server
```bash
apt install nginx
```
```